schema-tool/src/main/java/command/ConfigFileBasedCommand.java

```diff
@@ -0,0 +1,49 @@
+package command;
+
+import com.scalar.db.config.DatabaseConfig;
+import core.SchemaOperator;
+import java.io.FileInputStream;
+import java.util.HashMap;
+import java.util.concurrent.Callable;
+import java.util.logging.Logger;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Option;
+import picocli.CommandLine.Parameters;
+import utils.SchemaParser;
+
+@Command(name = "--config", description = "Using config file for Scalar DB")
+public class ConfigFileBasedCommand implements Callable<Integer> {
+
+  @Parameters(index = "0", description = "Path to config file of Scalar DB")
+  String configPath;
+
+  @Option(
+      names = {"-f", "--schema-file"},
+      description = "Path to schema json file",
+      required = true)
+  String schemaFile;
+
+  @Option(
+      names = {"-D", "--delete"},
+      description = "Delete tables",
+      defaultValue = "false")
+  Boolean deleteTables;
+
+  @Override
+  public Integer call() throws Exception {
+
+    Logger.getGlobal().info("Config path: " + configPath);
+    Logger.getGlobal().info("Schema path: " + schemaFile);
+
+    DatabaseConfig dbConfig = new DatabaseConfig(new FileInputStream(configPath));
+    SchemaOperator operator = new SchemaOperator(dbConfig);
+    SchemaParser schemaMap = new SchemaParser(schemaFile, new HashMap<String, String>());
```

Comment:
Using the `--config` mode, eventually it is not possible to set custom options ?

Reply:
If `DatabaseConfig` supports some options, then those options might be set as default when conducting the connection I think. Custom options can be set in the schema as well for each table instead as well

Reply:
Actually, some options are not supported in `DatabaseConfig` like `compaction-strategy` and `RU` related options. And I don't think we should support those options in `DatabaseConfig` because they are used only in schema creation. So one idea is, can we support a config file + custom options as follows:
```
# For Cassandra, the custom options in this case are -n and -R
$ ./build/install/schema-tool/bin/schema-tool --config <PATH_TO_CONFIG_FILE for Cassandra> -f schema.json -n NetworkTopologyStrategy -R 3

# For DynamoDB, the custom options in this case are -r and --no-scaling
$ ./build/install/schema-tool/bin/schema-tool --config <PATH_TO_CONFIG_FILE for DynamoDB> -f schema.json -r 3000 --no-scaling
```

What do you think?

Reply:
I think it is a good option. I'll add it